src/backend/base/langflow/components/git/gitfile.py

```diff
@@ -0,0 +1,230 @@
+import os
+import tempfile
+from contextlib import asynccontextmanager, suppress
+from typing import Any
+
+import aiofiles
+import anyio
+import git
+from git.exc import GitCommandError
+
+from langflow.custom import Component
+from langflow.io import DropdownInput, MessageTextInput, MultiselectInput, Output
+from langflow.schema import Data
+
+
+class GitFileComponent(Component):
+    display_name = "GitFile"
+    description = "Analyzes a Git repository and returns the content of selected files from specified branch"
+    icon = "GitLoader"
+
+    inputs = [
+        MessageTextInput(
+            name="repository_url",
+            display_name="Repository URL",
+            info="URL of the Git repository (e.g., https://github.com/username/repo)",
+            required=True,
+        ),
+        DropdownInput(
+            name="branch",
+            display_name="Branch",
+            info="Select a branch from the repository",
+            options=["Enter repository URL first"],
+            value="Enter repository URL first",
+            refresh_button=True,
+            required=True,
+        ),
+        MultiselectInput(
+            name="selected_files",
+            display_name="Select Files",
+            info="Select one or more files from the repository",
+            options=["Select branch first"],
+            value=[],
+            refresh_button=True,
+            required=True,
+        ),
+    ]
+
+    outputs = [
+        Output(display_name="Files Content", name="files_content", method="get_files_content"),
+    ]
+
+    def __call__(self, *args: Any, **kwargs: Any) -> Any:
+        """Override to handle test cases."""
```

Comment:
Can we avoid test code leaking into main code ?
Is it possible to mock inside the test instead ?